app.go

```diff
@@ -902,3 +902,13 @@ func ServerUserAgent(hostName string) string {
 	}
 	return "Go (" + serverSoftware + "/" + softwareVer + hostUAStr + ")"
 }
+
+// GetCollection returns the collection associated with the alias. If the application is in
+// single user mode then the collection associated with ID 1 (admin) is returned.
```

Comment:
As a first-time reader, the most confusing things about this func are:

* The app config's effects on the return values (that it affects the results at all)
* Conditions when the `alias` param is actually used

So I think the documentation could be improved by specifying that: The `alias` is only utilized in multi-user mode, otherwise the first **collection** in the database is returned (not necessarily the collection associated with a particular user). You might also mention that an empty `alias` value is acceptable.

Throughout the app, there are many different scenarios where we want to get a collection from the database. So I think this func name should be more descriptive, and explain _when_ someone should use it, e.g.:

* `GetCollectionFromURL`
* `GetCollectionFromAlias`
* `GetCollectionForView`
* etc.

Reply:
Thanks for the feedback. 

If the function needs to get a collection and getting this collection depends on whether there is or isn't a alias in the URL, then I think it'd be good to put all this logic in one place. 

> In single-user mode, no alias is given in the URL -- e.g. https://matt.writefreely.dev/ -- so we can't look up the collection from the URL alone. This is compared to multi-user mode, where the alias is in the URL, e.g. https://write.as/matt/, (matt is the alias).

I think `GetCollectionFromURL` would make a lot of sense. We could continue using the logic as rewritten but in all cases if the config is in single user mode then the `alias == ""` right? The function signature could be modified to take in the request and retrieve the alias from the URL/check the app for its user mode. 

Is the alias typically used for anything else other than getting the collection?